src/text_engine/interaction/Confirmation.java

```diff
@@ -0,0 +1,41 @@
+package text_engine.interaction;
+
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+
+import text_engine.characters.PlayerCharacter;
+import text_engine.interaction.actions.Action;
+import text_engine.items.GameEntity;
+
+public class Confirmation extends Interactive {
+
+    private final Action action;
+    private final String prompt;
+    private final Boolean defaultChoice;
+
+    /**
+     * Create and show a confirmation to the user with the given prompt. Will run the given {@link
+     * Action} if the player answers in the positive, or return to the given {@link Interactive} if
+     * the player answers in the negative.
+     *
+     * @param action        {@link Action} to run if the player responds in the positive
+     * @param prompt        to present to the player when asking for a response
+     * @param defaultChoice the default response of the player, if the player submits empty input. If
+     *                      {@code null}, an explicit response is required.
+     */
+    public Confirmation(Action action, String prompt, Boolean defaultChoice) {
+        this.action = action;
+        this.prompt = prompt;
+        this.defaultChoice = defaultChoice;
+    }
+
+    @Override
+    protected int respondToInteraction(PlayerCharacter actor, GameEntity from, BufferedReader reader,
+                                       PrintWriter writer) throws ExitToException {
+        if (ConsolePrompt.getChoiceBoolean(reader, writer, prompt, defaultChoice)) {
+            action.apply(actor);
+        }
+
+        return GoTo.PARENT;
+    }
+}
```

Comment:
This will be convenient for a lot of things. It can be used for exiting the game, since `action.apply(...)` can call `System.exit()` and the return statement will never be reached, or it can be used for generic confirmation dialogues like consuming an item, where execution should continue at the same place after the dialogue is done. This provides an easy and atomic way to do both.